RecoLocalTracker/SiStripClusterizer/plugins/SiStripClustersToLegacy.cc

```diff
@@ -0,0 +1,161 @@
+      // The number of clusters from x->nClusterCandidates() is an upper limit,
+      // the flag trueCluster then mask the real clusters.
+      // From Run: 386593 Event: 536278171 there are nClusterCandidates=112735 with
+      // 99863 real clusters (so 112735-99863 = 12872 clusters are masked out )
+      const uint32_t clusterCandidatesNb = clusters_onHost->nClusterCandidates();
+      const uint32_t goodClustersNb = clusters_onHost->candidateAcceptedPrefix(clustersArrSize - 1);
+
+      using out_t = edmNew::DetSetVector<SiStripCluster>;
+      auto output = std::make_unique<out_t>();
+      // output->reserve(nModulesWithClustersGuess, goodClustersNb);
+
+      // Debugging
+      // output->reserve(nModulesWithClustersGuess, 10);
+      // iEvent.put(siStripClustersSetVecPutToken_, std::move(output));
+      // return ;
+
+      // std::vector<uint8_t> adcs;
+      uint32_t clusterN = 0;
+      for (uint32_t i = 0; i < clusterCandidatesNb && (clusterN < goodClustersNb);) {
+        const auto detid = detIdArr[i];
+        // std::cout << "#fledm," << i << "," << detid << std::endl;
+        out_t::FastFiller record(*output, detid);
+
+        while (clusterN < goodClustersNb && detIdArr[i] == detid) {
+          if (candidateAcceptedArr[i]) {
+            const auto size = clusterSizeArr[i];
+            const auto firstStrip = firstStripArr[i];
+
+            const float barycenter = barycenterArr[i];
+            const float charge = chargeArr[i];
+
+            // std::vector<uint8_t> adcs;
+            // adcs.reserve(size);
+
+            // for (uint32_t j = 0; j < size; ++j) {
+            //   const auto index = clusterIndexArr[i] + j;
+            //   const auto adc = clusterAmplArr[index];
+            //   adcs.push_back(adc);
+            // }
+
+            const auto index = clusterIndexArr[i];
+            auto clusterAdc = clusterAmplArr.subspan(index, size);
+            std::vector<uint8_t> adcs(clusterAdc.begin(), clusterAdc.end());
+
+            // SiStripCluster(uint16_t firstStrip, std::vector<uint8_t>&& data, float barycenter, float charge)
+            record.push_back(SiStripCluster(firstStrip, std::move(adcs), barycenter, charge));
+            clusterN++;
+          }
+          i++;
+        }
+      }
+
+      if (edm::isDebugEnabled()) {
+        dumpClusters(output.get());
+      }
+      iEvent.put(siStripClustersSetVecPutToken_, std::move(output));
+    }
+
+    void dumpClusters(edmNew::DetSetVector<SiStripCluster>* detSetClusters) const;
+
+  private:
+    const edm::EDGetTokenT<SiStripClusterHost> siStripClustersToken_;
+    const edm::EDGetTokenT<SiStripDigiHost> siStripDigiToken_;
+    const edm::EDPutTokenT<edmNew::DetSetVector<SiStripCluster>> siStripClustersSetVecPutToken_;
+  };
+
+  void SiStripClustersToLegacy::dumpClusters(edmNew::DetSetVector<SiStripCluster>* detSetClusters) const {
+    int clustersAlloc = detSetClusters->dataSize();
+
+    std::ostringstream dumpMsg("");
```

Comment:
Nitpicking, feel free to ignore:
```suggestion
    std::ostringstream dumpMsg;
```